docs/use/affinity.md

```diff
@@ -0,0 +1,170 @@
+
+## Use cases for affinity
+
+Here are several detailed use cases for affinity that highlight its diverse applications and the numerous benefits it offers across various contexts.
+
+
+=== "Node affinity"
+    ### Regional scheduling
+
+    Let's consider a use case in which workloads are distributed based on performance requirements, fault tolerance, and regional specifications across designated zones or areas.
+
+    You need to run a workload in the `us-west2 region` for latency optimaztion and to meet specific compliance requirements.
+
+    ```sh
+    affinity:
+      nodeAffinity:
+        requiredDuringSchedulingIgnoredDuringExecution:
+        nodeSelectorTerms:
+        - matchExpressions:
+            - key: topology.kubernetes.io/region
+            operator: In
+            values:
+            - "us-west2"
+    ```
+    ??? info "What happens under the hood"
+        - It ensures that the pod is scheduled only on nodes located in the us-west-2 region, as defined by the `topology.kubernetes.io/region` node label.
+
+        - For the rule to be applicable, it is essential that the node possesses a label such as `topology.kubernetes.io/region: us-west-2`. If no nodes correspond to the specified label, the pod will remain in an unscheduled state.
+
+
+=== "Pods affinity"
+    ### Pods scheduled together
+
+    Let's consider a use case in which you want to to ensure that HAProxy pods should be scheduled to run on the same Kubernetes node.
+    
+        affinity:
+          podAffinity:
+            requiredDuringSchedulingIgnoredDuringExecution:
+            - labelSelector:
+            matchExpressions:
+                - key: app
+                operator: In
+                values:
+                - haProxy
+        topologyKey: "kubernetes.io/hostname"
+
+    ??? info "What happens under the hood"
+        - The pod will be scheduled only on nodes that already contain pods labeled with `app=haproxy`.
+
+        - If no nodes match, the pod will not be scheduled until a suitable node becomes available.
+
+
+=== "Pod anti-affinity"
+    ### Pods scheduled apart
+
+    Let's consider a use case in which you want to ensure that no HAProxy pods should be scheduled to run on the same Kubernetes node.
+
+
+        affinity:
+          podAntiAffinity:
+            requiredDuringSchedulingIgnoredDuringExecution:
+            - labelSelector:
+            matchExpressions:
+                - key: app
+                operator: NotIn
+                values:
+                - haproxy
+        topologyKey: "kubernetes.io/hostname"
+
+    ??? info "What happens under the hood"
+        - The pod will not be scheduled on nodes that contain pods labeled with `app=haproxy`.
+
+        - If no nodes match, the pod will not be scheduled until a suitable node becomes available.
```

Comment:
Why don't we use the default rule that we set as an example?
Simply setting `topologyKey: "kubernetes.io/hostname"` ensures that the scheduler will try to spread the different DB pods across different k8s nodes, ensuring better fault tolerance in case of node failure.

Reply:
We can have it explained here itself.

Reply:
Definitely. It will be more useful.